core/modules/qdisp/QdispPool.cc

```diff
@@ -201,17 +204,46 @@ std::string PriorityQueue::statsStr() {
 }
 
 
+QdispPool::QdispPool(int poolSize, int largestPriority, std::vector<int> const& maxRunSizes,
+                      std::vector<int> const& minRunningSizes) {
+    std::stringstream os;
+    os << "poolSize=" << poolSize
+       << " maxPriority=" << largestPriority << " maxRunSizes=" << util::prettyCharList(maxRunSizes)
+       << " minRunningSizes=" << util::prettyCharList(minRunningSizes);
+    if (poolSize < 1 || largestPriority < 0 || maxRunSizes.size() < static_cast<size_t>(largestPriority) + 1) {
+        LOGS(_log, LOG_LVL_ERROR, "QdispPool invalid paramater " << os.str());
+        throw std::invalid_argument(os.str());
+    }
+
+    LOGS(_log, LOG_LVL_INFO, "QdispPool creating " << os.str());
+    _prQueue = std::make_shared<PriorityQueue>(100, 1, 1); // default (lowest) priority.
+    for (unsigned int pri = 0; pri <= static_cast<unsigned int>(largestPriority); ++pri) {
```

Comment:
Is there a protection that this loop won't go "wild" and create thousands or millions of priority queues? Someone may just configure Qserv to set the highest priority to 10000000 (which is quit elegal) in the config file and bad things will happen here!.

Reply:
I added upper limits for poolSize and largestPriority.